### PR TITLE
DOC:  Create `class Extension` docstring and add it to documentation.

### LINKED ISSUE #14633: Add API documentation for np.distutils.core.Extension class
[The f2py docs mention](https://docs.scipy.org/doc/numpy/f2py/distutils.html) that the `numpy.distutils.core.Extension` class can be used, but as far as I can tell there is no documentation of this class either in the main numpy documentation or in the f2py documentation.

### PR DESCRIPTION
**Pull request by dstansby**
Fixes #14633. I have also

- Removed individual entries for everything in `misc_util` from the main page.
- Moved the `ConfigurationClass` doc down a section

### CODE REVIEW

**Comment by mattip** [comment]
docs build is failing.

File: doc/source/reference/distutils.rst [hunk 1]
```diff
@@ -23,37 +23,25 @@ information is available in the :ref:`distutils-user-guide`.
...
+   system_info.get_info
+   system_info.get_standard_file
+
 
+Configuration class
+===================
+
+.. currentmodule:: numpy.distutils
 
 .. class:: Configuration(package_name=None, parent_name=None, top_path=None, package_path=None, **attrs)
```
Comment (mattip) [inline doc/source/reference/distutils.rst]: this class lives in `numpy.distutils.misc_util`

**Comment by mattip** [comment]
Why did you remove the functions from `misc_util`? They are referenced from [here](https://numpy.org/devdocs/reference/distutils_guide.html#useful-functions-in-numpy-distutils-misc-util), it would be nice to turn those into links, but in any case they should be linked in somehow.

**Comment by dstansby** [comment]
I thought it didn't really fit with the rest of the page, and most of the individual functions aren't even documented. I added a link to the `misc_utils` module in the module table of contents, but could re-instate the individual links to functions further down the page?

**Comment by dstansby** [comment]
Yes, there's quite a lot that needs cleaning up with the distutils docs! For now I've added a new folder into which distutils `automodule` pages can go.

File: doc/source/reference/distutils.rst (hunk 1 above)
Comment (seberg) [reply doc/source/reference/distutils.rst]: Oh, thanks. somewhat thought there were two PRs failing.

Comment (charris) [reply doc/source/reference/distutils.rst]: Why are you removing all those other entries?

**Comment by dstansby** [comment]
I think this is good now - the new `distutils/misc_util` has all the functions in `misc_util` documented.

**Comment by mattip** [comment]
2 minor nits: 
- The f2py/distutils.rst page could use a reference directly to `numpy.distutils.Extension` now, which I think was the goal of #14633 
- The link to `distutils.misc_utils [here](https://9597-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/reference/distutils.html#module-numpy.distutils) looks a bit out of place. Is it because of the autosummary?

**Comment by rgommers** [comment]
Let's get this in, it's a significant improvement over the current docs. Thanks @dstansby!

**Pull request merged**